tests/qchem/test_tapering.py

```diff
@@ -952,3 +953,40 @@ def test_inconsistent_callable_ops(operation, op_wires, op_gen, message_match):
         taper_operation(
             operation, generators, paulixops, paulix_sector, wire_order, op_wires, op_gen
         )
+
+
+@pytest.mark.parametrize(
+    ("symbols", "geometry", "charge"),
+    [
+        (
+            ["H", "H"],
+            np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.40104295]], requires_grad=True),
+            0,
+        ),
+        (
+            ["He", "H"],
+            np.array(
+                [[0.0, 0.0, 0.0], [0.0, 0.0, 1.4588684632]],
+                requires_grad=True,
+            ),
+            1,
+        ),
+    ],
+)
+@pytest.mark.parametrize(
+    "max_size",
+    [2, 5],
+)
+def test_split_pauli_sentence(symbols, geometry, charge, max_size):
+    """Test that _split_pauli_sentence splits the PauliSentence objects into correct chunks."""
+
+    mol = qml.qchem.Molecule(symbols, geometry, charge)
+    hamiltonian = qml.qchem.diff_hamiltonian(mol)(geometry)
+
+    pauli_sentence = qml.pauli.pauli_sentence(hamiltonian)
+    split_sentence = {}
+
+    for ps in _split_pauli_sentence(pauli_sentence, max_size=max_size):
+        split_sentence = {**split_sentence, **ps}
```

Comment:
We should also check that `ps` is actually split into a term of the correct size:
```suggestion
    for ps in _split_pauli_sentence(pauli_sentence, max_size=max_size):
        assert len(ps) <= max_size
        split_sentence = {**split_sentence, **ps}
```